v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsSimpleIT.java

```diff
@@ -0,0 +1,242 @@
+    JsonArray ja = new JsonArray();
+    ja.add(jsObj);
+    String shardFileContents = ja.toString();
+    LOG.info("Shard file contents: {}", shardFileContents);
+    // -DartifactBucket has the bucket name
+    gcsClient.createArtifact("input/shard.json", shardFileContents);
+  }
+
+  private void uploadSessionFileToGcs() throws IOException {
+    gcsClient.uploadArtifact(
+        "input/session.json", Resources.getResource(sessionFileResourceName).getPath());
+  }
+
+  private void createSpannerMetadataDatabase() throws IOException {
+    spannerMetadataResourceManager =
+        SpannerResourceManager.builder("rr-meta-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String dummy = "create table t1(id INT64 ) primary key(id)";
+    spannerMetadataResourceManager.executeDdlStatement(dummy);
+    // needed to create separate metadata database
+    spannerMetadataDatabaseName = spannerMetadataResourceManager.getDatabaseId();
+  }
+
+  private void launchReaderDataflowJob() throws IOException {
+    // default parameters
+    Map<String, String> params =
+        new HashMap<>() {
+          {
+            put("sessionFilePath", getGcsPath("input/session.json"));
+            put("instanceId", spannerResourceManager.getInstanceId());
+            put("databaseId", spannerResourceManager.getDatabaseId());
+            put("spannerProjectId", PROJECT);
+            put("metadataDatabase", spannerMetadataResourceManager.getDatabaseId());
+            put("metadataInstance", spannerMetadataResourceManager.getInstanceId());
+            put("sourceShardsFilePath", getGcsPath("input/shard.json"));
+            put("changeStreamName", "allstream");
+            put("runIdentifier", "run1");
+            put("gcsOutputDirectory", getGcsPath("output"));
+          }
+        };
+
+    // Construct template
+    String jobName = PipelineUtils.createJobName(testName);
+    // /-DunifiedWorker=true when using runner v2
+    LaunchConfig.Builder options = LaunchConfig.builder(jobName, specPath);
+    options.setParameters(params);
+    options.addEnvironment("additionalExperiments", Collections.singletonList("use_runner_v2"));
+    // Run
+    jobInfo = launchTemplate(options, false);
+    assertThatPipeline(jobInfo).isRunning();
+  }
+
+  private void writeSpannerData() {
+    // Write a single record to Spanner for logical shard : testA
+    Mutation m =
+        Mutation.newInsertOrUpdateBuilder("Singers")
+            .set("SingerId")
+            .to(1)
+            .set("FirstName")
+            .to("FF")
+            .set("shardId")
+            .to("testA")
+            .build();
+    spannerResourceManager.write(m);
+  }
+
+  private void assertFileContentsInGCS() throws IOException, java.lang.InterruptedException {
+    List<Artifact> artifacts = null;
+    Thread.sleep(300000); // wait sufficiently for the file to be generated
```

Comment:
300 seconds seems very long. Can we do 30s wait and check in a loop for 10 times? and maybe assert content and size outside the loop

Reply:
It takes near about 2 or 3 minutes sometimes for the first file, per my observation. So I kept a 5 minute wait. If file is not generated by then - then surely something is wrong. In this case I want to throw exception. If we want to put in loop then I think it will be better to add a sleep for at-least 2 minutes and then do a loop to check if file has arrived. Does that look okay?

Reply:
Yeah. Sounds good

Reply:
Done